types/lockfile-info/index.d.ts

```diff
@@ -0,0 +1,15 @@
+// Type definitions for lockfile-info 1.0
+// Project: https://github.com/ljharb/lockfile-info
+// Definitions by: Jordan Harband <https://github.com/ljharb>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+interface LockfileInfo {
+    hasPackageJSON: boolean;
+    hasNodeModulesDir: boolean;
+    hasLockfile: boolean;
+    hasPackageLock: boolean;
+    hasShrinkwrap: boolean;
+    lockfileVersion: typeof NaN | 1 | 2 | 3; // https://github.com/microsoft/TypeScript/issues/47347
+}
+declare function lockfileInfo(cwd?: string): Promise<LockfileInfo>;
```

Comment:
```suggestion
/** @async */
declare function lockfileInfo(cwd?: string): Promise<LockfileInfo>;
```
https://jsdoc.app/tags-async.html
(TS definition file has no notion of async/await otherwise)